searchcore/src/vespa/searchcore/proton/server/proton.cpp

```diff
@@ -462,6 +462,15 @@ Proton::~Proton()
     LOG(debug, "Explicit destructor done");
 }
 
+void
+Proton::perform_pre_service_layer_shutdown_steps()
+{
+    _stateServer.reset();
```

Comment:
We need to reset _customComponentRootToken and _customComponentBindToken first, as these have reference to a JsonHandlerRepo owned by the StateServer, and will access this when destructed.

Reply:
In Proton::~Proton() we do `_executor.sync();` before any of this. I am not sure if we need something similar here. @toregge can you comment on that?

Reply:
That depends on what queued tasks can contain references to the structure being destroyed.  We should probably close the config fetcher, disable reconfig and sync the executor, to avoid new DocumentDB instances arriving after we've decided to take down the state server.

Reply:
I have moved config, token, state server and metrics shutdown into a dedicated method. Explicit executor sync is done in both places. PTAL